web/skins/classic/views/watch.php

```diff
@@ -20,57 +20,40 @@
 
 require_once('includes/Monitor.php');
 
-if ( !canView( 'Stream' ) )
-{
-    $view = "error";
-    return;
+if ( !canView( 'Stream' ) ) {
+  $view = 'error';
+  return;
 }
 
 // This is for input sanitation
 $mid = intval( $_REQUEST['mid'] ); 
 if ( ! visibleMonitor( $mid ) ) {
-    $view = "error";
-    return;
+  $view = 'error';
+  return;
 }
 
-$sql = 'SELECT C.*, M.* FROM Monitors AS M LEFT JOIN Controls AS C ON (M.ControlId = C.Id ) WHERE M.Id = ?';
 $monitor = new Monitor( $mid );
-#dbFetchOne( $sql, NULL, array( $_REQUEST['mid'] ) );
 
+#Whether to show the actual controls
 if ( isset($_REQUEST['showControls']) )
-    $showControls = validInt($_REQUEST['showControls']);
+  $showControls = validInt($_REQUEST['showControls']);
 else
-    $showControls = (canView( 'Control' ) && ($monitor->DefaultView() == 'Control'));
+  $showControls = canView( 'Control' ) && ($monitor->DefaultView() == 'Control');
 
+#Whether to show the controls button
 $showPtzControls = ( ZM_OPT_CONTROL && $monitor->Controllable() && canView( 'Control' ) );
 
 if ( isset( $_REQUEST['scale'] ) ) {
   $scale = validInt($_REQUEST['scale']);
 } else if ( isset( $_COOKIE['zmWatchScale'.$mid] ) ) {
   $scale = $_COOKIE['zmWatchScale'.$mid];
 } else {
-  $scale = reScale( SCALE_BASE, $monitor->DefaultScale, ZM_WEB_DEFAULT_SCALE );
+  $scale = reScale( SCALE_BASE, $monitor->DefaultScale(), ZM_WEB_DEFAULT_SCALE );
 }
 
 $connkey = generateConnKey();
 
-if ( ZM_WEB_STREAM_METHOD == 'mpeg' && ZM_MPEG_LIVE_FORMAT )
-{
-    $streamMode = "mpeg";
-    $streamSrc = $monitor->getStreamSrc( array( "mode=".$streamMode, "scale=".$scale, "bitrate=".ZM_WEB_VIDEO_BITRATE, "maxfps=".ZM_WEB_VIDEO_MAXFPS, "format=".ZM_MPEG_LIVE_FORMAT ) );
-}
-elseif ( canStream() )
-{
-    $streamMode = "jpeg";
-    $streamSrc = $monitor->getStreamSrc( array( "mode=".$streamMode, "scale=".$scale, "maxfps=".ZM_WEB_VIDEO_MAXFPS, "buffer=".$monitor->StreamReplayBuffer() ) );
-}
-else
-{
-    $streamMode = "single";
-    $streamSrc = $monitor->getStreamSrc( array( "mode=".$streamMode, "scale=".$scale ) );
-    Info( "The system has fallen back to single jpeg mode for streaming. Consider enabling Cambozola or upgrading the client browser.");
-}
-
+$streamMode = getStreamMode();
```

Comment:
Hi !

I cannot find this function defined. I also have an error when accessing /index.php?view=watch&mid=1 too, I cannot see the stream